src/ImageSharp/Formats/Jpeg/Components/Encoder/ComponentProcessor.cs

```diff
@@ -0,0 +1,233 @@
+                sourceRow.Slice(0, packedWidth).CopyTo(this.ColorBuffer.DangerousGetRowSpan(i / factors.Height));
+            }
+
+            static void SumVertical(Span<float> target, Span<float> source)
+            {
+                if (Avx.IsSupported)
+                {
+                    ref Vector256<float> targetVectorRef = ref Unsafe.As<float, Vector256<float>>(ref MemoryMarshal.GetReference(target));
+                    ref Vector256<float> sourceVectorRef = ref Unsafe.As<float, Vector256<float>>(ref MemoryMarshal.GetReference(source));
+
+                    // Spans are guaranteed to be multiple of 8 so no extra 'remainder' steps are needed
+                    nint count = source.Length / Vector256<float>.Count;
+                    for (nint i = 0; i < count; i++)
+                    {
+                        Unsafe.Add(ref targetVectorRef, i) = Avx.Add(Unsafe.Add(ref targetVectorRef, i), Unsafe.Add(ref sourceVectorRef, i));
+                    }
+                }
+                else
+                {
+                    ref Vector<float> targetVectorRef = ref Unsafe.As<float, Vector<float>>(ref MemoryMarshal.GetReference(target));
+                    ref Vector<float> sourceVectorRef = ref Unsafe.As<float, Vector<float>>(ref MemoryMarshal.GetReference(source));
+
+                    nint count = source.Length / Vector<float>.Count;
+                    for (nint i = 0; i < count; i++)
+                    {
+                        Unsafe.Add(ref targetVectorRef, i) += Unsafe.Add(ref sourceVectorRef, i);
+                    }
+
+                    ref float targetRef = ref MemoryMarshal.GetReference(target);
+                    ref float sourceRef = ref MemoryMarshal.GetReference(source);
+                    for (nint i = count * Vector<float>.Count; i < source.Length; i++)
+                    {
+                        Unsafe.Add(ref targetRef, i) += Unsafe.Add(ref sourceRef, i);
+                    }
+                }
+            }
+
+            static void SumHorizontal(Span<float> target, int factor)
+            {
+                if (Avx2.IsSupported)
+                {
+                    ref Vector256<float> targetRef = ref Unsafe.As<float, Vector256<float>>(ref MemoryMarshal.GetReference(target));
+
+                    // Ideally we need to use log2: Numerics.Log2((uint)factor)
+                    // but division by 2 works just fine in this case
+                    // because factor value range is [1, 2, 4]
+                    // log2(1) == 1 / 2 == 0
+                    // log2(2) == 2 / 2 == 1
+                    // log2(4) == 4 / 2 == 2
+                    int haddIterationsCount = (int)((uint)factor / 2);
+                    int length = target.Length / Vector256<float>.Count;
+                    for (int i = 0; i < haddIterationsCount; i++)
+                    {
+                        length /= 2;
+                        for (int j = 0; j < length; j++)
```

Comment:
```suggestion
                        for (nuint j = 0; j < length; j++)
```
also for `indexLeft` and `indexRight`.